pkg/controller/windowsmachineconfig/tracker/tracker.go

```diff
@@ -0,0 +1,247 @@
+package tracker
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift/windows-machine-config-bootstrapper/tools/windows-node-installer/pkg/types"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// StoreName is the name of the ConfigMap used to track Windows worker node information
+const StoreName = "wmco-node-tracker"
+
+var log = ctrllog.Log.WithName("tracker")
+
+// nodeRecord holds information about a single node in the cluster. This is the information that will be placed in the
+// ConfigMap with the node's cloud ID as the key to get the record.
+type nodeRecord struct {
+	// CredSecret holds name of the secret for each Windows node managed by the WMCO
+	CredSecret string `json:"credSecret"`
+	// IPAddress of the Windows node managed by the WMCO
+	IPAddress string `json:"ipAddress"`
+	// Drain indicates if the Windows node has to be drained before removing the Windows VMs from cluster
+	Drain bool `json:"drain"`
+}
+
+// Tracker is used to track the instance information
+type Tracker struct {
+	// windowsVMs is a map of Windows VMs to be tracked by WMCO
+	windowsVMs map[types.WindowsVM]bool
+	// operatorNS is the name of namespace where the operator is running.
+	operatorNS string
+	//k8sclientset is the kubernetes client set
+	k8sclientset *kubernetes.Clientset
+	// nodeRecords is a map of instanceID as key and marshalled nodeRecord object
+	nodeRecords map[string][]byte
+}
+
+// NewTracker initializes and returns a tracker object
+func NewTracker(k8sclientset *kubernetes.Clientset, windowsVMs map[types.WindowsVM]bool) (*Tracker, error) {
+	if k8sclientset == nil {
+		return nil, fmt.Errorf("cannot instantiate tracker without k8s client")
+	}
+	if windowsVMs == nil {
+		return nil, fmt.Errorf("cannot instantiate tracker with a nil windowsVMs slice")
+	}
+
+	// Get the namespace the operator is currently deployed in.
+	operatorNS, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get operator namespace")
+	}
+
+	return &Tracker{
+		windowsVMs:   windowsVMs,
+		operatorNS:   operatorNS,
+		k8sclientset: k8sclientset,
+	}, nil
+}
+
+// WindowsVMs sets Windows VMs to be tracked by WMCO
```

Comment:
s/Windows VMs/windowsVMs to indicate this is just a setter for a field in the struct.